src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -1350,12 +1484,48 @@ DatabaseShardImp::getCache(std::uint32_t seq)
     return std::make_pair(pCache, nCache);
 }
 
-std::uint64_t
-DatabaseShardImp::available() const
+bool
+DatabaseShardImp::sufficientStorage(
+    std::uint32_t numShards,
+    bool isHistorical,
+    std::lock_guard<std::mutex> const&) const
 {
     try
     {
-        return boost::filesystem::space(dir_).available;
+        std::vector<std::uint64_t> capacities;
+
+        // Get the available storage for each historical path
+        if (isHistorical && !historicalPaths_.empty())
+        {
+            std::for_each(
+                historicalPaths_.begin(),
+                historicalPaths_.end(),
+                [&capacities](boost::filesystem::path const& path) {
+                    capacities.push_back(
+                        boost::filesystem::space(path).available);
+                });
```

Comment:
A range for loop feels simpler for this. I.e.
```
            capacities.reserve(historicalPaths_.size());
            for (auto const& p : historicalPaths_)
                capacities.push_back(boost::filesystem::space(p).available);
```
Note, even if we end up keeping the `for_each`, we should still `reserve`

Reply:
Added reserve.